lib/winston-graylog2.js

```diff
@@ -28,86 +26,65 @@ let getMessageLevel = (function() {
   };
 })();
 
-/**
- * Preparing metadata for graylog
- * If we send a javascript `Error` object
- * to gray log we'll end up with the
- * `[object Object]` string.
- * To have our infos we need to get the stack out of the error.
- * Here we remap metadata to handle this kind of situation
- *
- * @param  {Object} meta
- * @param  {Object} staticMeta
- * @return {Object}
- */
-function prepareMeta(meta, staticMeta) {
-  meta = meta || {};
-
-  if (meta instanceof Error) {
-    meta = {error: meta.stack};
-  } else if (typeof meta === 'object') {
-    meta = _.mapValues(meta, function(value) {
-      if (value instanceof Error) {
-        return value.stack;
-      }
-
-      return value;
+/** Class representing the Graylog2 Winston Transport */
+class Graylog2 extends Transport {
+  /**
+   * Create the transport
+   * @param {Object} options - The options for configuring the transport.
+   */
+  constructor(options) {
+    super(options);
+
+    options = options || {};
+    this.graylog = options.graylog;
+    if (!this.graylog) {
+      this.graylog = {
+        servers: [
+          {
+            host: 'localhost',
+            port: 12201,
+          },
+        ],
+      };
+    }
+
+    this.name = options.name || 'graylog2';
+    this.exceptionsLevel = options.exceptionsLevel || 'not-default';
+
+    this.silent = options.silent || false;
+    this.handleExceptions = options.handleExceptions || false;
+
+    this.graylog2Client = new Graylog2Client(this.graylog);
+    this.staticMeta = options.staticMeta || {};
+
+    this.graylog2Client.on('error', function(error) {
+      console.error('Error while trying to write to graylog2:', error);
     });
   }
 
-  meta = _.merge(meta, staticMeta);
-
-  return meta;
-}
-
-let Graylog2 = (winston.Graylog2 = function(options) {
-  winston.call(this, options);
-
-  options = options || {};
-  this.graylog = _.get(options, 'graylog');
-  if (!this.graylog) {
-    this.graylog = {
-      servers: [
-        {
-          host: 'localhost',
-          port: 12201,
-        },
-      ],
-    };
+  /**
+   * Log a message to Graylog2.
+   *
+   * @param {Object} info - An object containing the `message` and `info`.
+   * @param {function} callback - Winston's callback to itself.
+   */
+  log(info, callback) {
+    const {message, level, ...metadata} = info;
+    const meta = Object.assign(metadata.metadata, this.staticMeta);
```

Comment:
So the `...` rest parameter will dump everything remaining in the `info` object aside from `message` and `level` back into `metadata`. I initially wrote it like that because at the time I wasn't sure how the additional stuff was being added into `info`. I think you're right though. I should just be able to remove the rest parameter and we should be good.